src/App.tsx

```diff
@@ -15,35 +16,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT,
+  ALPHABETICALLY,
+  LENGTH,
+}
+
+interface SortingConditionals {
+  sortingType: SortType;
+  isReversed: boolean;
+}
+
+const sortingGoodsFunction = (
+  goodsArray: string[],
+  { sortingType, isReversed }: SortingConditionals,
+) => {
+  const goodsArrayCopy = [...goodsArray];
+
+  switch (sortingType) {
+    case SortType.ALPHABETICALLY:
+      goodsArrayCopy.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.LENGTH:
+      goodsArrayCopy.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed === true) {
+    return goodsArrayCopy.reverse();
+  } else {
+    return goodsArrayCopy;
+  }
+};
```

Comment:
The name of the function `sortingGoodsFunction` does not start with a verb, which could make it unclear what the function does. Consider renaming it to start with a verb, such as `sortGoods` to follow the [CODE STYLE] checklist item.